### PR TITLE
User Notes

### PR DESCRIPTION
**Pull request by florian-sabonchi**
**Description**
Provide a description of what your changes do.

**Issue**
Closes #XX

**Additional Notes**
Add any other context about the pull request here.

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: moderation/user_notes/cog.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+from PyDrocsid.cog import Cog
+from PyDrocsid.converter import UserMemberConverter
+from PyDrocsid.database import db, filter_by, select
+from PyDrocsid.embeds import send_long_embed
+from PyDrocsid.emojis import name_to_emoji
+from PyDrocsid.translations import t
+from cogs.library.moderation.user_notes.models import UserNote
+from cogs.library.moderation.user_notes.permissions import UserNotePermission
```
Comment (Defelo) [inline moderation/user_notes/cog.py]: use relative imports here

File: moderation/user_notes/cog.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+from discord import Embed
+from discord.ext import commands
+from discord.ext.commands import Context, UserInputError, guild_only, CommandError
+from library.PyDrocsid.command import docs
```
Comment (Defelo) [inline moderation/user_notes/cog.py]:
```suggestion
from PyDrocsid.command import docs
```

File: moderation/user_notes/cog.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+from .colors import Colors
+from ...contributor import Contributor
+from ...pubsub import send_to_changelog
+
+t = t.user_notes
```
Comment (Defelo) [inline moderation/user_notes/cog.py:21]:
```suggestion
tg = t.g
t = t.user_notes
```

File: moderation/user_notes/cog.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+
+class UserNoteCog(Cog, name="User notes"):
```
Comment (Defelo) [inline moderation/user_notes/cog.py]:
```suggestion
class UserNotesCog(Cog, name="User Notes"):
```

File: moderation/user_notes/cog.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    CONTRIBUTORS = [Contributor.Florian]
+
+    @commands.group()
+    @guild_only()
+    @UserNotePermission.read.check
```
Comment (Defelo) [inline moderation/user_notes/cog.py]:
```suggestion
    @UserNotePermission.read.check
    @guild_only()
```

File: moderation/user_notes/cog.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+from ...contributor import Contributor
+from ...pubsub import send_to_changelog
+
+t = t.user_notes
+
+
+class UserNoteCog(Cog, name="User notes"):
+    CONTRIBUTORS = [Contributor.Florian]
+
+    @commands.group()
```
Comment (Defelo) [inline moderation/user_notes/cog.py]: add command aliases

File: moderation/user_notes/cog.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    @UserNotePermission.read.check
+    @docs(t.description)
+    async def user_note(self, ctx: Context):
+
+        if ctx.invoked_subcommand is None:
+            raise UserInputError
+
+    @user_note.command(name="add")
+    @UserNotePermission.write.check
+    async def add_user_note(self, ctx: Context, member: UserMemberConverter, *, message: str):
```
Comment (Defelo) [inline moderation/user_notes/cog.py]: add command description

File: moderation/user_notes/cog.py [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+            member=member.id,
+            author=ctx.author.id,
+            message=message,
+        )
+        await ctx.message.add_reaction(name_to_emoji["white_check_mark"])
+        await send_to_changelog(ctx.guild, t.new_note(member.mention, f"<@{ctx.author.id}>", message))
+
+    @user_note.command(name="remove")
+    @UserNotePermission.write.check
+    async def remove_user_note(self, ctx: Context, message_id: int):
```
Comment (Defelo) [inline moderation/user_notes/cog.py]: add command description

File: moderation/user_notes/cog.py [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+        user_note: UserNote = await db.get(UserNote, message_id=message_id)
```
Comment (Defelo) [inline moderation/user_notes/cog.py]:
```suggestion
        user_note: Optional[UserNote] = await db.get(UserNote, message_id=message_id)
```

File: moderation/user_notes/cog.py [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+        if not user_note:
+            raise CommandError(t.note_not_found)
+        await db.delete(user_note)
```
Comment (Defelo) [inline moderation/user_notes/cog.py:89]: use `confirm` from PyDrocsid to create a simple confirmation dialog

File: moderation/user_notes/cog.py [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+        await ctx.message.add_reaction(name_to_emoji["white_check_mark"])
+        await send_to_changelog(
+            ctx.guild,
+            t.removed_note(f"<@{user_note.member}>", f"<@{user_note.author}>", user_note.message),
+        )
+
+    @user_note.command(name="show")
+    async def show_user_note(self, ctx: Context, member: UserMemberConverter):
```
Comment (Defelo) [inline moderation/user_notes/cog.py]: add command description

File: moderation/user_notes/cog.py [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+        embed = Embed(title=t.user_notes, colour=Colors.user_notes)
+        async for note in await db.stream(select(UserNote).filter_by(member=member.id)):
+            embed.add_field(name=t.id, value=note.message_id, inline=True)
+            embed.add_field(name=t.message, value=note.message, inline=True)
+            embed.add_field(name=t.timestamp, value=note.timestamp.strftime("%d.%m.%Y %H:%M:%S"), inline=True)
+            embed.add_field(name=t.author, value=f"<@{note.author}>", inline=True)
+        await send_long_embed(ctx, embed)
```
Comment (Defelo) [inline moderation/user_notes/cog.py]:
```suggestion
        await send_long_embed(ctx, embed, paginate=True)
```

File: moderation/user_notes/cog.py [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+        )
+
+    @user_note.command(name="show")
+    async def show_user_note(self, ctx: Context, member: UserMemberConverter):
+        embed = Embed(title=t.user_notes, colour=Colors.user_notes)
+        async for note in await db.stream(select(UserNote).filter_by(member=member.id)):
+            embed.add_field(name=t.id, value=note.message_id, inline=True)
+            embed.add_field(name=t.message, value=note.message, inline=True)
+            embed.add_field(name=t.timestamp, value=note.timestamp.strftime("%d.%m.%Y %H:%M:%S"), inline=True)
+            embed.add_field(name=t.author, value=f"<@{note.author}>", inline=True)
```
Comment (Defelo) [inline moderation/user_notes/cog.py]: put the entire note into one embed field. you can use the timestamp as the name (as in the `ulog` command)

File: moderation/user_notes/translations/en.yml [hunk 14]
```diff
@@ -0,0 +1,12 @@
+user_notes: "User Notes"
+id: "Message ID"
+message: "Message"
+timestamp: "Timestamp"
+author: "Author"
+description: "Mange notes for users"
```
Comment (Defelo) [inline moderation/user_notes/translations/en.yml]: use a `commands` dictionary to store the command descriptions

File: moderation/user_notes/cog.py [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+    @commands.group()
+    @guild_only()
+    @UserNotePermission.read.check
+    @docs(t.description)
+    async def user_note(self, ctx: Context):
+
+        if ctx.invoked_subcommand is None:
+            raise UserInputError
+
+    @user_note.command(name="add")
```
Comment (Defelo) [inline moderation/user_notes/cog.py]: add command aliases

File: moderation/user_notes/cog.py [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+    async def add_user_note(self, ctx: Context, member: UserMemberConverter, *, message: str):
+        await UserNote.create(
+            member=member.id,
+            author=ctx.author.id,
+            message=message,
+        )
+        await ctx.message.add_reaction(name_to_emoji["white_check_mark"])
+        await send_to_changelog(ctx.guild, t.new_note(member.mention, f"<@{ctx.author.id}>", message))
+
+    @user_note.command(name="remove")
```
Comment (Defelo) [inline moderation/user_notes/cog.py]: add command aliases

File: moderation/user_notes/cog.py [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+        if not user_note:
+            raise CommandError(t.note_not_found)
+        await db.delete(user_note)
+        await ctx.message.add_reaction(name_to_emoji["white_check_mark"])
+        await send_to_changelog(
+            ctx.guild,
+            t.removed_note(f"<@{user_note.member}>", f"<@{user_note.author}>", user_note.message),
+        )
+
+    @user_note.command(name="show")
```
Comment (Defelo) [inline moderation/user_notes/cog.py]: add command aliases

**Review by Defelo — CHANGES_REQUESTED** [review]

**Review by Defelo — APPROVED** [review]

**Pull request merged**